modules/dynamic-operator/plugin/postgresql/operator.go

```diff
@@ -0,0 +1,380 @@
+
+func (p *PostgresqlService) Empty(req commons.PluginRequest) *commons.PluginResponse {
+	p.logger.Debug("call Empty")
+	object, err := p.ToUnstructured(&postgresv1.Postgresql{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      req.Name,
+			Namespace: req.Namespace,
+		},
+	})
+	if err != nil {
+		return &commons.PluginResponse{
+			Error: err,
+		}
+	}
+	return &commons.PluginResponse{
+		Object: object,
+	}
+}
+
+func (p *PostgresqlService) merge(object *postgresv1.Postgresql, req commons.PluginRequest) error {
+	object.Spec.NumberOfInstances = req.Replicas
+	object.Spec.Volume.Size = req.VolumeSize
+	object.Spec.PgVersion = req.Version
+
+	for k, v := range req.Parameters {
+		switch k {
+		case "resources":
+			result := &postgresv1.Resources{}
+			err := p.FromUnstructured(v.(map[string]interface{}), result)
+			if err != nil {
+				return errors.New(fmt.Sprintf("cannot convert %v to postgresv1.Resources", v))
+			}
+			object.Spec.Resources = *result
+		default:
+			// unknown parameter
+			p.logger.Error("unknown parameter", "key", k, "value", v)
+		}
+	}
+	return nil
+}
+
+func (p *PostgresqlService) IsReady() bool {
+	switch p.service.Status.PostgresClusterStatus {
+	case postgresv1.ClusterStatusCreating, postgresv1.ClusterStatusUpdating, postgresv1.ClusterStatusUnknown:
+		return false
+	case postgresv1.ClusterStatusAddFailed, postgresv1.ClusterStatusUpdateFailed, postgresv1.ClusterStatusSyncFailed, postgresv1.ClusterStatusInvalid:
+		return false
+	case postgresv1.ClusterStatusRunning:
+		return true
+	default:
+		return false
+	}
+}
+
+func (p *PostgresqlService) Status(req commons.PluginRequest) *commons.PluginResponse {
+	p.logger.Debug("call Status")
+	object := &postgresv1.Postgresql{}
+	err := p.FromUnstructured(req.Object.UnstructuredContent(), object)
+	if err != nil {
+		p.logger.Error(err.Error())
+		return &commons.PluginResponse{
+			Error: err,
+		}
+	}
+
+	isReady := p.IsReady()
+	p.logger.Info("isReady", "result", isReady)
+	return &commons.PluginResponse{
+		IsReady: isReady,
+	}
+}
+
+func (p *PostgresqlService) ForUpdate(req commons.PluginRequest) *commons.PluginResponse {
+	p.logger.Debug("call ForUpdate")
+	object := &postgresv1.Postgresql{}
+	err := p.FromUnstructured(req.Object.UnstructuredContent(), object)
+	if err != nil {
+		p.logger.Error(err.Error())
+		return &commons.PluginResponse{
+			Error: err,
+		}
+	}
+
+	p.merge(object, req)
+	p.logger.Info("ForUpdate", "object", object)
+
+	response, err := p.ToUnstructured(object)
+	if err != nil {
+		p.logger.Error(err.Error())
+		return &commons.PluginResponse{
+			Error: err,
+		}
+	}
+
+	return &commons.PluginResponse{
+		Object: response,
+	}
+}
+
+func (p *PostgresqlService) ForCreate(req commons.PluginRequest) *commons.PluginResponse {
+	p.logger.Debug("call ForCreate")
+	object := p.Init(req)
+	p.merge(object, req)
+
+	reps, err := p.ToUnstructured(object)
+	if err != nil {
+		p.logger.Error(err.Error())
+		return &commons.PluginResponse{
+			Error: err,
```

Comment:
> encoding/gob has been frozen for ages now and we aren't adding more features/support to it, thus I shall close this issue. So long encoding/gob, you served us well!

https://github.com/golang/go/issues/23340

:(
So now it does not work correctly with type "error", that is to say does not work with interfaces.